### PR TITLE
ISSUE-264 | Replace bullet to n_plus_one

### LINKED ISSUE #264: [Gem] Replace bullet with n_plus_one_contol
Gem - https://github.com/palkan/n_plus_one_control
Add to all specs
Add info to readme

### PR DESCRIPTION
**Pull request by alexyndr**
[ISSUE-264](https://github.com/rubygarage/boilerplate/issues/264)

### Description

- Remove bullet gem
- Add n_plus_one_control gem
- Add few n+1 tests to requests

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project

### CODE REVIEW

File: spec/requests/api/v1/users/account/profiles_spec.rb [hunk 1]
```diff
@@ -35,5 +35,14 @@
         include_examples 'renders uri query errors'
       end
     end
+
+    context 'N+1', :n_plus_one do # rubocop:disable RSpec/ContextWording
```
Comment (dzemlianoi-double) [inline spec/requests/api/v1/users/account/profiles_spec.rb]: why disabling here?
why context - not 'describe'?

File: spec/requests/api/v1/users/verifications_spec.rb [hunk 2]
```diff
@@ -36,5 +36,14 @@
         end
       end
     end
+
+    context 'N+1', :n_plus_one do # rubocop:disable RSpec/ContextWording
```
Comment (dzemlianoi-double) [inline spec/requests/api/v1/users/verifications_spec.rb]: Have you checked that it's raising if it's n+1?

**Comment by dzemlianoi-double** [comment]
Have you checked other api requests?

File: spec/requests/api/v1/users/verifications_spec.rb (hunk 2 above)
Comment (alexyndr) [reply spec/requests/api/v1/users/verifications_spec.rb]: Yes. I injected one n+1 query in code, and get this result in falling specs:
`Expected to make the same number of queries, but got:
         5 for N=2
         6 for N=3
       Unmatched query numbers by tables:
         users (SELECT): 3 != 4
`

File: spec/requests/api/v1/users/account/profiles_spec.rb (hunk 1 above)
Comment (alexyndr) [reply spec/requests/api/v1/users/account/profiles_spec.rb]: i just followed examples, you right better will be `describe`

File: spec/requests/api/v1/users/verifications_spec.rb (hunk 2 above)
Comment (alexyndr) [reply spec/requests/api/v1/users/verifications_spec.rb]: I add specs for all `GET requests`

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@
 abort('The Rails environment is running in production mode!') if Rails.env.production?
 
 require 'rspec/rails'
+require 'n_plus_one_control/rspec'
```
Comment (dzemlianoi-double) [inline spec/rails_helper.rb]: separate config file please

File: spec/requests/api/v1/users/account/profiles_spec.rb [hunk 4]
```diff
@@ -6,9 +6,10 @@
   let!(:account) { create(:account, :with_user) }
   let(:headers) { { Authorization: auth_header(account) } }
   let(:params) { {} }
+  let(:profile_url) { '/api/v1/users/account/profile' }
```
Comment (dzemlianoi-double) [inline spec/requests/api/v1/users/account/profiles_spec.rb:9]: Do we need to check not just GET actions. Could we have a situation, something like

```
Post.all.each do |post|
  post.category.update(name: 'test')
end
```

**Review by dzemlianoi-double — CHANGES_REQUESTED** [review]

**Pull request merged**